apps/frontend/core-dapp/src/components/Toastify.tsx

```diff
@@ -0,0 +1,86 @@
+import { Flex, Icon, spacingIncrement } from 'prepo-ui'
+import { ToastContainer, toast as _toast, ToastOptions as ToastifyOptions } from 'react-toastify'
+import styled from 'styled-components'
+import Link from './Link'
+
+type ToastOptions = {
+  description?: React.ReactNode
+  link?: string
+} & ToastifyOptions
+
+const Title = styled.p`
+  color: ${({ theme }): string => theme.color.secondary};
+  font-weight: ${({ theme }): number => theme.fontWeight.semiBold};
+`
+
+const TitleLink = styled(Link)`
+  color: ${({ theme }): string => theme.color.secondary};
+  display: flex;
+  text-decoration: none;
+  width: max-content;
+  > div > div {
+    color: ${({ theme }): string => theme.color.neutral5};
+  }
+  :hover {
+    color: ${({ theme }): string => theme.color.secondary};
+    opacity: 0.8;
+    text-decoration: none;
+    > div > div {
+      color: ${({ theme }): string => theme.color.neutral5};
+    }
+  }
+`
+
+const DescriptionWrapper = styled.div`
+  color: ${({ theme }): string => theme.color.neutral3};
+  font-size: ${({ theme }): string => theme.fontSize.sm};
+`
+
+const toastFunction = (title: string, _options: ToastOptions = {}): void => {
+  const { description, link, ...options } = _options
+
+  const titleComponent =
+    link === undefined ? (
+      <Title>{title}</Title>
+    ) : (
+      <TitleLink href={link} target="_blank">
+        <Flex gap={8} justifyContent="flex-start" width="max-content">
+          <Title>{title}</Title>
+          <div>
+            <Icon
+              name="arrow-up-right"
+              height={spacingIncrement(16)}
+              width={spacingIncrement(16)}
+            />
+          </div>
+        </Flex>
+      </TitleLink>
+    )
+
+  _toast(
+    <Flex flexDirection="column" justifyContent="flex-start" alignItems="flex-start" gap={8}>
+      {titleComponent}
+      {description !== undefined && <DescriptionWrapper>{description}</DescriptionWrapper>}
+    </Flex>,
+    options
+  )
+}
+
+type ToastParameters = Parameters<typeof toastFunction>
+
+export const toast = {
+  error: (...[title, options]: ToastParameters): void =>
+    toastFunction(title, { ...options, type: 'error' }),
+  success: (...[title, options]: ToastParameters): void =>
+    toastFunction(title, { ...options, type: 'success' }),
+}
```

Comment:
```suggestion
export const toast = {
  error: (title: string, options: ToastOptions): void =>
    toastFunction(title, { ...options, type: 'error' }),
  success: (title: string, options: ToastOptions): void =>
    toastFunction(title, { ...options, type: 'success' }),
}
```

It's a bit annoying but, I think that it's much cleaner to simply repeat the types.